### DISCUSSION TITLE
filter operator ?= not working

### DISCUSSION THREAD

**Discussion by m-2k** [pocketbase/pocketbase#4347] [Q&A] [answered]
table `projects` with multirelation field `members` refers to `users` collection
PB/jsSDK v0.21.3 (v0.19.0 same)

```js
const id = "w9ic8dwejwdvv4g"
const filter = p.filter("members ?= {:id}", { id }) // members ?= 'w9ic8dwejwdvv4g'
await p.collection("projects").getFullList({ filter }) // -> []
```

```url
http://localhost:8090/api/collections/projects/records/?filter=(members?=%22w9ic8dwejwdvv4g%22)
=>
{
    "page": 1,
    "perPage": 30,
    "totalItems": 0,
    "totalPages": 0,
    "items": []
}
```

but

```url
http://localhost:8090/api/collections/projects/records/?filter=(members?~%22w9ic8dwejwdvv4g%22)
=>
{
    "page": 1,
    "perPage": 30,
    "totalItems": 1,
    "totalPages": 1,
    "items": [
        {
            "collectionId": "istuum1n3p76xzi",
            "collectionName": "projects",
            "created": "2024-02-13 13:00:47.216Z",
            "id": "et99nj9yrzq3dek",
            "members": [
                "w9ic8dwejwdvv4g"
            ],
            "name": "proj-1",
   ...
}
```

**Comment by ganigeorgiev** [comment, marked as answer]
`members` refer to the raw value, which in case of a multiple `relation` field is a serialized json array (eg. `["id1","id2"]`).

The `members ~ "id"` operator works because it applies a `LIKE` string matching expression against the raw serialized  json array.

The `members ?= "id"` doesn't work because it compares the raw serialized json array against that single id string.

To target a specific relation field you can use dot-notation: `members.id ?= "id"`.

**Reply by m-2k** [reply]
is this in the documentation?
